client/components/resource/ResourceList.tsx

```diff
@@ -63,65 +62,153 @@ const ResourceListTitle = styled(PageTitle)`
   white-space: nowrap;
 `
 
-const RightSpaceWrapper = styled.div`
-  margin-right: auto;
-`
+type sortPatternType = (first: Resource, second: Resource) => number
 
 interface ResourceListHeaderProps {
   onSearch: (search: string) => any
+  changeSortPattern: (sortPatternType) => void
+  slug: string
 }
 
-interface ResourceListTagsProps {
-  tags: Array<Tag>
-}
+const Row = styled.div`
+  display: flex;
+  align-items: center;
+`
+
+const SortDropdownSelect = styled.select`
+  padding: 4px;
+  border-radius: 16px;
+  font-size: 16px;
+  background: white;
+  font-family: ${fontInter};
+  &:hover {
+    background: ${colours.neutralLight1};
+  }
+  height: 32px;
+  @media ${largerThan(width.tablet)} {
+    padding: 8px;
+    border-radius: 24px;
+    height: 48px;
+  }
+
+  &:focus {
+    outline: none;
+    border-color: ${colours.primary2};
+  }
+`
+
+const SortByLabel = styled.label`
+  margin-right: 10px;
+`
+
+const SortDropdown = ({
+  slug,
+  changeSortPattern,
+}: {
+  slug: string
+  changeSortPattern: (sortPatternType) => void
+}) => {
+  let criteria = RatingCriteria[slug]
+  let sortingDefinitions = {
+    alphabetical: (first: Resource, second: Resource) => {
+      return first.name.localeCompare(second.name) // sort from least (a) to most (z)
+    },
+    'number of ratings': (first: Resource, second: Resource) => {
+      return second.totalReviews - first.totalReviews // sort from most to least
+    },
+  }
+
+  criteria.map((item) => {
+    sortingDefinitions[item] = (first: Resource, second: Resource) => {
+      return second.averageRating[item] - first.averageRating[item] // sort from most to least
+    }
+  })
+
+  criteria = criteria.map((item) => capitalizeFirstLetter(item))
+  criteria = ['Alphabetical', 'Number of Ratings', ...criteria]
```

Comment:
In general, it's best to avoid mutation like this, assigning back to the same variable and to keep things clean by creating new variables.

```
const criteria = RatingCriteria[slug]
const normalizedCriteria = criteria.map((criterion) => capitalizeFirstLetter(criterion)).unshift(['Alphabetical', 'Number of Ratings']
```

```